variables.tf

```diff
@@ -0,0 +1,10 @@
+variable "s3_account_settings" {
+  description = "Object of inputs for S3 account settings"
+  type = object({
+    account_id              = optional(string)
+    block_public_acls       = optional(bool)
+    block_public_policy     = optional(bool)
+    ignore_public_acls      = optional(bool)
+    restrict_public_buckets = optional(bool)
+  })
+}
```

Comment:
Let's take this even further here, and prepare for multiple resources in this module. Each resource gets its own nested object. And also set the "secure" defaults.

```
variable "s3_account" {
  description = "Object of inputs for S3 account-level settings"
  type = object({
    public_access_block = optional({
      account_id              = optional(string)
      block_public_acls       = optional(bool, true)
      block_public_policy     = optional(bool, true)
      ignore_public_acls      = optional(bool, true)
      restrict_public_buckets = optional(bool, true)
    }, {})
  })
}
```